src/Optimiser.cpp

```diff
@@ -325,16 +335,21 @@ double Optimiser::optimise_partition(vector<MutableVertexPartition*> partitions,
   }
 
   // Make sure the resulting communities are called 0,...,r-1
-  // where r is the number of communities.
+  // where r is the number of communities. The exception is fixed
+  // nodes which should keep the numbers of the original communities
   q = 0.0;
   vector<size_t> membership = MutableVertexPartition::renumber_communities(partitions);
```

Comment:
Here we should then renumber using `partitions[0]->renumber_communities(original_fixed_memberships)`. We could leave the original statement as well, so that the communities will in principle first be ordered, and then be renumbered to follow the original fixed nodes. In other words, we obtain

    vector<size_t> membership = MutableVertexPartition::renumber_communities(partitions);
    partitions[0]->set_membership(membership);
    membership = partitions[0]->renumber_communities(original_fixed_membership);

The subsequent `for` loop can then start at `layer = 1` of course.

Reply:
done

Reply:
I just realised that the last line of my proposal would not set the membership of course, so the `for` loop should still start at `layer = 0`.

Reply:
Correct, let me fix this

Reply:
done